hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/EightToNineUpgradeHandler.java

```diff
@@ -117,6 +124,18 @@ public UpgradeDowngrade.TableConfigChangeSet upgrade(HoodieWriteConfig config,
           metaClient.getIndexDefinitionPath(),
           indexMetadataOpt.get(),
           metaClient.getTableConfig().getTableVersion());
+      //Validate index columns
+      TableSchemaResolver tableSchemaResolver = new TableSchemaResolver(metaClient);
+      try {
+        List<String> partitionsToDrop = mdtPartitionsToDrop(indexMetadataOpt, tableSchemaResolver.getTableAvroSchema());
+        if (!partitionsToDrop.isEmpty()) {
+          try (BaseHoodieWriteClient writeClient = upgradeDowngradeHelper.getWriteClient(config, context)) {
+            writeClient.dropIndex(partitionsToDrop);
+          }
+        }
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
```

Comment:
I think the column stats index can follow the same upgrade and downgrade flow as secondary index where V1 column stats index can be updated after table upgrade from v8 to v9; during downgrade, V2 column stats index is dropped.